### PR TITLE
🐛Don't save story ads to history

### PR DESCRIPTION
**Pull request by calebcordry**
Ads do not persist after a nav to out-link. Therefore when returning to a page you cannot switch back to that ad.

Remove the logic that saves the ad to history.

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story.js [hunk 1]
```diff
@@ -936,6 +938,10 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   setHistoryStatePageId_(pageId) {
+    // Never save ad pages to history as they are unique to each visit.
+    if (pageId.slice(0, AD_PAGE_ID_PREFIX.length) === AD_PAGE_ID_PREFIX) {
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story.js]: Hmm... Can we instead get page by id and check `isAd`?

Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story.js]: Sure, its slower though. Not sure if it matters

Comment (newmuis) [reply extensions/amp-story/0.1/amp-story.js]: If we get by id using the story function (not querying the DOM) I wouldn't expect it to be slow enough to matter, but I'm not in love with hardcoding the assumption of the ads page ids here

Comment (calebcordry) [reply extensions/amp-story/0.1/amp-story.js]: Fair enough, I will change it. However we are in control of the creation of this id, and are guaranteeing it in [other places](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/amp-story-ads.md#tracking)

Comment (newmuis) [reply extensions/amp-story/0.1/amp-story.js]: If we just need this for tracking, maybe we should expose an analytics variable for whether the page is an ad?  It just feels like id is not very semantic

**Pull request merged**